### ISSUE TITLE
Arbitrary rust code errors

### ISSUE THREAD

**Issue by la10736** [la10736/rstest#16]
Should span eventually error message at the code source.

Now for
```rust
#[rstest_parametrize(
condition,
case(r(r#"vec![1,5,7].contains(23)"#))
)]
fn arbitrary(condition: bool) {
    assert!(condition);
}
```
We receive
```
error[E0308]: mismatched types
  --> src/main.rs:7:1
   |
7  | / #[rstest_parametrize(
8  | | condition,
9  | | case(r(r#"vec![1,5,7].contains(23)"#))
10 | | )]
   | |__^ expected &{integer}, found integral variable
   |
   = note: expected type `&{integer}`
              found type `{integer}`
help: consider borrowing here
   |
7  | &#[rstest_parametrize(
8  | condition,
9  | case(r(r#"vec![1,5,7].contains(23)"#))
10 | )]
   |
```
Here is quite complicated to understand where the error is (should be `&23`) but in a more complicated example (more cases and wired code) should be real impossible.

The error should point at least to arbitrary code.

Question: why `CaseArg::from(t).respan(a.span())` that we already use doesn't do the job?
 Should we think to use `quote_span!` to redirect eventually errors at least to case line?

**Comment by la10736** [comment]
Closed via 5c30a9604b8ae220e86c7a5c66

**Comment by la10736** [comment]
Wrong commit
